theano/tensor/basic.py

```diff
@@ -4372,7 +4372,7 @@ def infer_shape(self, node, ishapes):
                     if ele == -1:
                         requ[i] = missing
             elif crit == 1:  # we reshape to -1
-                requ = [mul(*ishapes[0])]
+                requ = [mul(*ishapes[0])] if ishapes[0] else [1]
```

Comment:
Ok. We could change the interface, to make it return a constant 1 of floatX dtype, but I'm not sure it is a good idea, so I won't push it. But if you think it is a good idea, tell it.